topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -473,6 +475,33 @@ public void setTopicCount(int topics) {
                 action.execute().onComplete(actionResult -> {
                     LOGGER.debug("{}: Executing handler for action {} on topic {}", logContext, action, lockName);
                     action.result = actionResult;
+
+                    String keytag = namespace + ":" + "KafkaTopic" + "/" + key.asKubeName().toString();
+                    Optional<Meter> metric = metrics.meterRegistry().getMeters()
+                            .stream()
+                            .filter(meter -> meter.getId().getName().equals(METRICS_PREFIX + "resource.state") &&
+                                    meter.getId().getTags().contains(Tag.of("kind", "KafkaTopic")) &&
+                                    meter.getId().getTags().contains(Tag.of("name", action.topic.getMetadata().getName())) &&
```

Comment:
Tested with this change and it seems to work. Please do some manual testing with renaming the topics.
```suggestion
                                                                        meter.getId().getTags().contains(Tag.of("name", action.topic == null ? key.asKubeName().toString() : action.topic.getMetadata().getName())) &&
```

Reply:
Yeah. I am testing them out